src/nwbinspector/checks/time_series.py

```diff
@@ -129,3 +129,14 @@ def check_resolution(time_series: TimeSeries):
         return InspectorMessage(
             message=f"'resolution' should use -1.0 or NaN for unknown instead of {time_series.resolution}."
         )
+
+
+@register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=TimeSeries)
+def check_rate_is_not_zero(time_series: TimeSeries):
+    if time_series.data is not None:
+        data_shape = get_data_shape(time_series.data) 
```

Comment:
Another suggestion related to the other complexity one; this is more of a style, but it is one that does help readability when you get into the habit

Early exit conditions are easier to understand than later structural returns

What I mean by that, I'll just show here

```python
if time_series.data is None:
    return
data_shape = get_data_shape(time_series.data)   # no need for an else statement either
```

This also helps with a minor issue where `data_shape` wouldn't even be a defined variable when/if the program hits lines 141 below

Reply:
Ok, thanks! I am including your correction.